src/eradiate/srf_tools.py

```diff
@@ -0,0 +1,824 @@
+            "# wavelength": f"{ni}",
+        },
+        "Final": {
+            "Lower wavelength": f"{w_min_f:~}",
+            "Upper wavelength": f"{w_max_f:~}",
+            "# wavelength": f"{nf}",
+        },
+        "Difference": {
+            "Lower wavelength": f"{(w_min_f-w_min_i):~}",
+            "Upper wavelength": f"{(w_max_f-w_max_i):~}",
+            "# wavelength": f"{nf - ni}",
+        },
+    }
+
+
+def summarize(
+    srf: t.Union[PathLike, xr.Dataset],
+    filtered: t.Union[PathLike, xr.Dataset],
+) -> Table:
+    """
+    Summarise the spectral response function filtering operation.
+
+    Parameters
+    ----------
+    srf: Dataset
+        Initial spectral response function data set.
+
+    filtered: Dataset
+        Filtered spectral response function data set.
+
+    Returns
+    -------
+    :class:`~rich.table.Table`
+        Summary table.
+    """
+    # convert inputs
+    srf = converters.load_dataset(srf)
+    filtered = converters.load_dataset(filtered)
+
+    summary = filtering_summary(srf=srf, filtered=filtered)
+    table = Table(
+        title="Filtering summary",
+        box=rich.box.HEAVY_HEAD,
+        show_footer=True,
+    )
+
+    columns = {
+        "SRF": "left",
+        "Lower wavelength": "center",
+        "Upper wavelength": "center",
+        "# wavelength": "center",
+    }
+
+    footer = {
+        **{"SRF": "Difference"},
+        **{
+            name: summary["Difference"][name]
+            for name in ["Lower wavelength", "Upper wavelength", "# wavelength"]
+        },
+    }
+
+    for name, justification in columns.items():
+        table.add_column(name, justify=justification, footer=footer[name])
+
+    for key in summary:
+        row_items = [key] + list(summary[key].values())
+
+        if key == "Difference":
+            pass
+        else:
+            table.add_row(*row_items)
+
+    return table
+
+
+def trim(srf: t.Union[PathLike, xr.Dataset]) -> xr.Dataset:
+    """
+    Trim all leading zeros except last and all trailing zeros except first.
+
+    Parameters
+    ----------
+    srf: path-like, Dataset
+        Data set to trim.
+
+    Returns
+    -------
+    Dataset
+        Trimmed data set.
+    """
+    ds = converters.load_dataset(srf)
+
+    # trim
+    wsize = ds.srf.values.size
+    fsize = np.trim_zeros(ds.srf.values, trim="f").size
+    bsize = np.trim_zeros(ds.srf.values, trim="b").size
+    istart = wsize - fsize - 1 if wsize > fsize else 0
+    istop = bsize if bsize < wsize else wsize - 1
+    trimmed = ds.isel(w=range(istart, istop + 1))
+
+    # update history attribute
+    previous_history = ds.attrs["history"] + "\n"
+    utcnow = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+    author = f"eradiate, version {__version__}"
+    history_attr = f"{utcnow} - trimmed data set - {author}"
+    trimmed.attrs.update({"history": f"{previous_history}{history_attr}"})
+
+    return trimmed
+
+
+def save_filtered_data(
```

Comment:
How about just `save()` and overall a less specific scoping? I understand that it's hardly necessary here, but in general, it's good practice to have the scope right. This function can actually write any dataset to the disk, its real features are the dry-run and logging.